src/lib/high_level/data_types.rs

```diff
@@ -196,6 +197,115 @@ pub trait Encryptable {
     fn as_bytes(&self) -> Option<[u8; 16]> {
         self.value().encode_lizard()
     }
+
+    /// Encodes an arbitrary byte array into one or more Encryptables
+    /// Uses PKCS#7 style padding where the padding byte value equals the number of padding bytes
+    fn from_bytes_padded(data: &[u8]) -> Vec<Self>
+    where
+        Self: Sized,
+    {
+        if data.is_empty() {
+            return vec![];
+        }
+
+        let mut result = Vec::new();
+
+        // Process all full blocks, that do not need padding
+        // Initialize the last block with the padding value
+        // Copy remaining data if there is any
+        for i in 0..(data.len() / 16) {
+            let start = i * 16;
+            // This is safe, as we know that the slice is 16 bytes long
+            result.push(Self::from_bytes(
+                &data[start..start + 16].try_into().unwrap(),
+            ));
+        }
+
+        let remaining = data.len() % 16;
+        let padding_byte = (16 - remaining) as u8;
+
+        let mut last_block = [padding_byte; 16];
+
+        if remaining > 0 {
+            last_block[..remaining].copy_from_slice(&data[data.len() - remaining..]);
+        }
+
+        result.push(Self::from_bytes(&last_block));
+
```

Comment:
We do not check for incorrectly padded ciphertexts here... Maybe we should do that? Or rather not, because it just complicates our logic (and we don't offer any integrity whatsoever)

Reply:
Hmm It wouldn't hurt to add it but I would say it is not needed.